mkosi

```diff
@@ -507,7 +507,7 @@ def attach_image_loopback(args: argparse.Namespace, raw: Optional[IO[str]]):
         with complete_step('Detaching image file'):
             run(["losetup", "--detach", loopdev], check=True)
 
-def partition(loopdev: str, partno: Optional[int]) -> Optional[str]:
+def partition(loopdev: str, partno: int) -> str:
```

Comment:
What happens if partno is None?

Reply:
At runtime, nothing new. Type annotation simply tells Mypy to report errors if `partno` is anything but an int. Since at present it is obtained from `args.swap_partno` (for example) and that has a type of `Any` it is not type checked further.

I have a follow-up patch that makes this strictly typed, that is `args.swap_partno` is no longer `Any` but `Optional[int]`. Then the places that check for `if args.swap_partno is None: return' allow Mypy to infer that the more narrow type applies and thus get the correct `int` type.

Reply:
A follow up to this is in https://github.com/systemd/mkosi/pull/273 - it allows Mypy to prove that the type is correct in this fragment now.

Reply:
But right below, there is a check `if partno is None`, which is in disagreement with the declared type `partno: int`. I think the code inside the function should agree with the declared type.

Reply:
That's correct, I didn't notice it, it should be removed along with the introduction of the type annotation.